DatadogCore/Tests/Datadog/Core/Upload/DataUploadWorkerTests.swift

```diff
@@ -166,109 +166,105 @@ class DataUploadWorkerTests: XCTestCase {
-                XCTFail("Wrong command is sent!")
-            }
-        }
+        let initialUploadDelay = 0.01
+        let delay = DataUploadDelay(
+            performance: UploadPerformanceMock(
+                initialUploadDelay: initialUploadDelay,
+                minUploadDelay: 0,
+                maxUploadDelay: 1,
+                uploadDelayChangeRate: 0.01
+            )
+        )
 
         // When
         writer.write(value: ["k1": "v1"])
 
-        let server = ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 500)))
-        let httpClient = URLSessionClient(session: server.getInterceptedURLSession())
-
-        let dataUploader = DataUploader(
-            httpClient: httpClient,
-            requestBuilder: FeatureRequestBuilderMock()
-        )
         let worker = DataUploadWorker(
             queue: uploaderQueue,
             fileReader: reader,
-            dataUploader: dataUploader,
+            dataUploader: DataUploaderMock(uploadStatus: .mockWith(needsRetry: true)),
             contextProvider: .mockAny(),
             uploadConditions: DataUploadConditions.alwaysUpload(),
-            delay: mockDelay,
+            delay: delay,
             featureName: .mockAny()
         )
 
         // Then
-        server.waitFor(requestsCompletion: 1)
-        waitForExpectations(timeout: 1, handler: nil)
+        wait(until: { [uploaderQueue] in
+            uploaderQueue.sync {
+                delay.current > initialUploadDelay
+            }
+        }, andThenFulfill: delayChangeExpectation)
+        wait(for: [delayChangeExpectation])
         worker.cancelSynchronously()
     }
 
     func testWhenBatchSucceeds_thenIntervalDecreases() {
         let delayChangeExpectation = expectation(description: "Upload delay is decreased")
-        let mockDelay = MockDelay { command in
-            if case .decrease = command {
-                delayChangeExpectation.fulfill()
-            } else {
-                XCTFail("Wrong command is sent!")
-            }
-        }
-
+        let initialUploadDelay = 0.05
+        let delay = DataUploadDelay(
+            performance: UploadPerformanceMock(
+                initialUploadDelay: initialUploadDelay,
+                minUploadDelay: 0,
+                maxUploadDelay: 1,
+                uploadDelayChangeRate: 0.01
+            )
+        )
         // When
         writer.write(value: ["k1": "v1"])
 
-        let server = ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 200)))
-        let httpClient = URLSessionClient(session: server.getInterceptedURLSession())
-
-        let dataUploader = DataUploader(
-            httpClient: httpClient,
-            requestBuilder: FeatureRequestBuilderMock()
-        )
         let worker = DataUploadWorker(
             queue: uploaderQueue,
             fileReader: reader,
-            dataUploader: dataUploader,
+            dataUploader: DataUploaderMock(uploadStatus: .mockWith()),
```

Comment:
How do we configure "When Batch Succeeds" in this setup now? It was `.mockResponseWith(statusCode: 200)` before, but it is not clear right now.

Reply:
it's based `needsRetry` property, which is by default set to false in the mock. I made it explicit now.